merlin/models/tf/loader.py

```diff
@@ -618,3 +623,42 @@ def sample_batch(
     if not include_targets:
         return inputs
     return inputs, targets
+
+
+def get_default_hvd_seed_fn(seed=None):
+    """
+    Generate consistent dataloader shuffle seeds across workers
+    Reseeds each worker's dataloader each epoch to get fresh a shuffle
+    that's consistent across workers.
+    """
+    if HAS_GPU:
+        try:
+            import cupy
+        except ImportError:
```

Comment:
In what case is it possible to have a gpu but not cupy? And is the import error message correct? I see that you only need cupy if the GPU is available... Why cant we default to numpy in the case where GPU is available but not cupy?

Reply:
Good point. Updated.